linkerd/app/outbound/src/http/endpoint.rs

```diff
@@ -39,6 +40,7 @@ impl<C> Outbound<C> {
                 .push_on_response(svc::MapErrLayer::new(Into::<Error>::into))
                 .check_service::<T>()
                 .into_new_service()
+                .push(peer_proxy_errors::PeerProxyErrors::layer())
```

Comment:
:+1: for the renaming

Reply:
Is there any particular reason this has to exist on the endpoint stack?

In https://github.com/linkerd/linkerd2-proxy/pull/1183/commits/c03b0493589bea172d26c373d52047c0037cbbfc, I've moved it into the server stack which gives us the advantage of not having to worry about retries and, more importantly, eliminates the need to add a clienthandle onto all of our endpoint-stack tests. I think this move may fit a bit nicer as then this logic can live right next to the other error-handling module